e2e-tests/main_test.go

```diff
@@ -0,0 +1,114 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"os/exec"
+	"strconv"
+	"testing"
+	"time"
+)
+
+func runCommandWithOutput(cmd *exec.Cmd) (string, error) {
+	var stdout, stderr bytes.Buffer
+	cmd.Stdout = &stdout
+	cmd.Stderr = &stderr
+
+	err := cmd.Run()
+	if err != nil {
+		return "", fmt.Errorf("error running command: %v\nstdout: %s\nstderr: %s", err, stdout.String(), stderr.String())
+	}
+
+	return stdout.String(), nil
+}
+
+// From the output of the AbciInfo command, extract the latest block height.
+// The json bytes should look e.g. like this:
+// {"jsonrpc":"2.0","id":1,"result":{"response":{"data":"interchain-security-p","last_block_height":"2566","last_block_app_hash":"R4Q3Si7+t7TIidl2oTHcQRDNEz+lP0IDWhU5OI89psg="}}}
+func ExtractHeightFromInfo(jsonBytes []byte) (int, error) {
+	// Use a generic map to represent the JSON structure
+	var data map[string]interface{}
+
+	if err := json.Unmarshal(jsonBytes, &data); err != nil {
+		return 0, fmt.Errorf("Failed to unmarshal JSON %s \n error was %v", string(jsonBytes), err)
+	}
+
+	// Navigate the map and use type assertions to get the last_block_height
+	result, ok := data["result"].(map[string]interface{})
+	if !ok {
+		return 0, fmt.Errorf("Failed to navigate abci_info output structure trying to access result: json was %s", string(jsonBytes))
+	}
+
+	response, ok := result["response"].(map[string]interface{})
+	if !ok {
+		return 0, fmt.Errorf("Failed to navigate abci_info output structure trying to access response: json was %s", string(jsonBytes))
+	}
+
+	lastBlockHeight, ok := response["last_block_height"].(string)
+	if !ok {
+		return 0, fmt.Errorf("Failed to navigate abci_info output structure trying to access last_block_height: json was %s", string(jsonBytes))
+	}
+
+	return strconv.Atoi(lastBlockHeight)
+}
+
+// Tests happy path functionality for ABCI Info.
+func TestABCIInfo(t *testing.T) {
```

Comment:
Why is this test under `main_test.go`? Could we have it under a more appropriate location (e.g., under `abci_client`)?

Reply:
Since this is the only testing infra so far, I think I prefer keeping everything here until we need to split as the file gets bigger